src/main/java/com/google/sps/servlets/NewGameServlet.java

```diff
@@ -75,22 +75,43 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
       System.out.println(uid);
 
     // Get a reference to document alovelace
-    CollectionReference colRef = firestoreDb.collection("games");
+    CollectionReference gamesCol = firestoreDb.collection("games");
 
     // Prepare data to be inserted
-    Map<String, Object> data = new HashMap<>();
-        data.put("creator", uid);
+    Map<String, Object> gameData = new HashMap<>();
+        gameData.put("creator", uid);
+        gameData.put("title", gameTitle);
 
-    DocumentReference docRef = colRef.document();
-    // Asynchronously write data
-    ApiFuture<WriteResult> resultFirebase = docRef.set(data);
+    DocumentReference gameDoc = gamesCol.document();
+    gameDoc.set(gameData);
 
-    // result.get() blocks on response
-    try {
-      System.out.println("Update time : " + resultFirebase.get().getUpdateTime());
-    } catch(Exception e) {
-      e.printStackTrace();
-    }
+    // Prepare data to be inserted
+    Map<String, Object> questionData = new HashMap<>();
+        questionData.put("title", question);
+
+    CollectionReference questionCol = gameDoc.collection("questions");
+    DocumentReference questionDoc = questionCol.document();
+
+    questionDoc.set(questionData);
+
+
+    CollectionReference answerCol = questionDoc.collection("answers");
+    DocumentReference correctAns = answerCol.document();
+    DocumentReference wrongAns = answerCol.document();
```

Comment:
nit. I dont think you need this lines you can just access as 
`DocumentReference ans = questionDoc.collection("answers").document();`
And set both maps to ans, I think it would be the same thing

Reply:
Each time the `.document()` function its called, it generates a new document on the database with a unique key.